src/Microsoft.AspNetCore.Routing.Abstractions/LinkGenerationTemplate.cs

```diff
@@ -76,8 +76,8 @@ public abstract class LinkGenerationTemplate
         public abstract string GetUri(
             HttpContext httpContext,
             object values,
-            string scheme = default,
-            HostString? host = default,
+            string scheme,
+            HostString host,
```

Comment:
So the one thing that made me uncomfortable here was changing from `HostString?` to `HostString`. I think it's OK though because it's never valid to provide an empty host string. Therefore `default(HostString)` is OK to use as a sentinel value. But it's not a breaking change to make parameters optional in the future. 

For reference `PathString? basePath` is nullable because it is valid to have an 'empty' base path. We want to know the difference between 'empty' and unspecified.